upup/pkg/fi/nodeup/nodetasks/issue_cert.go

```diff
@@ -120,11 +124,28 @@ func (i *IssueCert) AddFileTasks(c *fi.ModelBuilderContext, dir string, name str
 }
 
 func (e *IssueCert) Run(c *fi.Context) error {
+	// Skew the certificate lifetime by up to 30 days based on information about the generating node.
+	// This is so that different nodes created at the same time have the certificates they generated
+	// expire at different times, but all certificates on a given node expire around the same time.
```

Comment:
Oh!  Now I get why you used the network interfaces - so that the skew is constant for a given machine!